assets/src/components/mapPage.tsx

```diff
@@ -57,70 +65,298 @@ const ToggleMobileDisplayButton = ({
+  const topLeft = new Point(445, 0)
+  const innerBounds = new Bounds(topLeft, mapContainerBounds.getBottomRight())
+  // The "new center" is the offset between the two bounding boxes centers
+  const offset = innerBounds
+    .getCenter()
+    .subtract(mapContainerBounds.getCenter())
+
+  const targetZoom = 16
+  const targetPoint = map
+      // Project the target point into screenspace for the target zoom
+      .project(points[0], targetZoom)
+      // Offset the target point in screenspace to move the center of the map
+      // to apply the padding to the center
+      .subtract(offset),
+    // convert the target point to worldspace from screenspace
+    targetLatLng = map.unproject(targetPoint, targetZoom)
+
+  // Zoom/Pan center of map to offset location in worldspace
+  map.setView(targetLatLng, targetZoom)
+}
+
+const useFollowingStateWithSelectionLogic = (
+  selectedVehicleId: string | null,
+  selectedVehicleRef: Vehicle | null
+) => {
+  const state = useInteractiveFollowerState(),
+    { setShouldFollow } = state
+
+  // when the selected vehicle ID and last api do or don't reference the same
+  // vehicle
+  const vehicleIdDiffers = selectedVehicleId !== selectedVehicleRef?.id
+  useEffect(() => {
+    // Only update the shouldFollow state once the cached value agrees with
+    // the selection state.
+    // Otherwise the follower may try to center on stale data from the
+    // previous selection before `useVehicleForId` resolves to it's next value.
+    if (selectedVehicleId !== null && vehicleIdDiffers === false) {
+      setShouldFollow(true)
+    }
+  }, [vehicleIdDiffers, setShouldFollow, selectedVehicleId])
+  return state
+}
+
+const MapDisplay = ({
+  selectedVehicleId,
+  setSelectedVehicle,
+  showVpc,
+}: {
+  selectedVehicleId: VehicleId | null
+  setSelectedVehicle: React.Dispatch<VehicleOrGhost | null>
+  showVpc: boolean
+}) => {
+  const deleteSelection = useCallback(() => {
+    setSelectedVehicle && setSelectedVehicle(null)
+  }, [setSelectedVehicle])
+
+  const stations = useStations()
+
+  const { socket } = useContext(SocketContext)
+
+  const selectedVehicleRef = useVehicleForIdCached(socket, selectedVehicleId),
+    { routeId = null, tripId = null } = selectedVehicleRef || {}
+  const shapes = useTripShape(tripId)
+
+  const position =
+    (selectedVehicleRef &&
+      isVehicle(selectedVehicleRef) && [
+        vehicleToLeafletLatLng(selectedVehicleRef),
+      ]) ||
+    []
+
+  const state = useFollowingStateWithSelectionLogic(
+    selectedVehicleId,
+    selectedVehicleRef
+  )
 
+  return (
+    <BaseMap
+      vehicles={[]}
+      allowStreetView={true}
+      stopCardDirection={selectedVehicleRef?.directionId}
+      includeStopCard={true}
+      stations={stations}
+      shapes={selectedVehicleRef?.isShuttle ? [] : shapes}
+      stateClasses={FollowerStatusClasses(state.shouldFollow)}
+    >
+      <>
+        {showVpc && selectedVehicleRef && isVehicle(selectedVehicleRef) && (
+          <>
```

Comment:
question (non-blocking): Are these empty tags needed?

Reply:
nope! I actually cleaned this up when implementing the VPC ghost card https://github.com/mbta/skate/pull/1895
I was considering cherry picking that commit into this PR

Reply:
Cherry picked into [`c536580` (#1891)](https://github.com/mbta/skate/pull/1891/commits/c53658007572b47cc69963b14d988ae50cde0875)